pkg/controller/sparkapplication/controller.go

```diff
@@ -1001,3 +1006,19 @@ func (c *Controller) hasApplicationExpired(app *v1beta2.SparkApplication) bool {
 
 	return false
 }
+
+// Clean up batch scheduling config if applicable.
+func (c *Controller) cleanUpBatchScheduling(oldApp, newApp *v1beta2.SparkApplication) error {
+	if needScheduling, scheduler := c.shouldDoBatchScheduling(newApp); needScheduling {
+		state := newApp.Status.AppState.State
+		// If new app state is completed or failed, no more app is running,
+		// and only once needs to clean up on completed or failed
+		if (state == v1beta2.CompletedState || state == v1beta2.FailedState) &&
```

Comment:
I suggest moving the state check to the caller of this method.

Reply:
thanks, I moved it to caller